lib/Controller/MessagesController.php

```diff
@@ -314,32 +314,49 @@ public function getThread(int $id): JSONResponse {
 	 * @NoAdminRequired
 	 * @TrapError
 	 *
-	 * @param int $id
+	 * @param int[] $ids
 	 * @param int $destFolderId
 	 *
 	 * @return JSONResponse
 	 *
 	 * @throws ClientException
 	 * @throws ServiceException
 	 */
-	public function move(int $id, int $destFolderId): JSONResponse {
+	public function move(array $ids, int $destFolderId): JSONResponse {
+		$uids = [];
+
 		try {
-			$message = $this->mailManager->getMessage($this->currentUserId, $id);
+			// Get the source and destination account/mailbox
+			$message = $this->mailManager->getMessage($this->currentUserId, $ids[0]);
 			$srcMailbox = $this->mailManager->getMailbox($this->currentUserId, $message->getMailboxId());
 			$dstMailbox = $this->mailManager->getMailbox($this->currentUserId, $destFolderId);
 			$srcAccount = $this->accountService->find($this->currentUserId, $srcMailbox->getAccountId());
 			$dstAccount = $this->accountService->find($this->currentUserId, $dstMailbox->getAccountId());
+
+			// Get uid of all messages provided and make sure all messages come from the same mailbox
+			foreach ($ids as $id) {
+				$message = $this->mailManager->getMessage($this->currentUserId, $id);
```

Comment:
```suggestion
				$message = $this->mailManager->getMessage($this->currentUserId, (int) $id);
```

because we can't rely on the correct type as far as I know

Reply:
Yes, but we have a discussion with @kesselb about changins that part of the code.

I've proposed that the frontend sends a deletion batch like this:

```
batch = {
   $mailboxId => {
        $messageId_1,
        $messageId_2,
   },
  $mailboxId_2 => {
        $messageId_3,
        $messageId_4,
        ...
   },
   ...
}
```

But, I think @kesselb noted that this batch job must also contains the $userId

Reply:
I've updated my deletion batch algorithm => https://github.com/nextcloud/mail/pull/4249/commits/f5880a7e714dc47d62663406cc5a9e14f184f17e

It should be compatible with PI (I couldn't test thoroughly because I don't have multiple accounts on my NC boxes)

Could you have a look?

If you find it good, maybe we should use it for the move() function too